crates/counter-agg/src/lib.rs

```diff
@@ -0,0 +1,258 @@
+use time_weighted_average::tspoint::TSPoint;
+use serde::{Deserialize, Serialize};
+use regression::{XYPair, RegressionSummary};
+
+pub mod regression;
+pub mod range;
+mod tests;
+
+#[derive(Debug, PartialEq)]
+pub enum CounterError{
+    OrderError,
+    BoundsInvalid,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
```

Comment:
Perhaps a comment here or at the top of the file explaining a bit better what these are and how they work (i.e. track monotonically increasing sensors which need to handle resets).